server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2274,8 +2274,8 @@ private ShardLongFieldRange determineShardLongFieldRange(String fieldName) {
             return ShardLongFieldRange.UNKNOWN; // no mapper service, no idea if the field even exists
         }
         final MappedFieldType mappedFieldType = mapperService().fieldType(fieldName);
-        if (mappedFieldType instanceof DateFieldMapper.DateFieldType == false) {
-            return ShardLongFieldRange.UNKNOWN; // field missing or not a date
+        if (mappedFieldType instanceof DateFieldMapper.DateFieldType == false || mappedFieldType.name().equals(fieldName) == false) {
```

Comment:
this fix is technically not necessary: the change below on the search side is enough to fix the issue, because we no longer look for the range if timestamp is an alias. Yet it seems wrong to mark the range EMPTY in that case, hence I think it makes sense to correct this side too.

Reply:
Once the index metadata contains a range with `org.elasticsearch.index.shard.IndexLongFieldRange#isComplete()` being `true`, it shouldn't change again, and it should reach that state after starting all the shards in the index the first time. But I know this area has moved around quite a bit since I last touched it and there was some confusion about `EMPTY` vs `UNKNOWN` in the past (e.g. around https://github.com/elastic/elasticsearch/pull/106252) whose resolution I didn't follow, so I can't be sure that it does exactly what I think it should do :)

Reply:
> does it main that it is a problem if it goes e.g. from EMPTY to UNKNOWN

That certainly sounds incorrect to me. `EMPTY` is supposed to mean "we are certain that no shards of this index will match any range query against this field and can therefore skip these shards entirely on every such query" whereas `UNKNOWN` is supposed to mean "we do not know what the timestamp range of this field is so cannot do any coordinator-node-side skipping". I can't see how we can go from being certain that skipping is correct to being unsure.

Reply:
I think the problem is that we previously (incorrectly) returned and thus set `EMPTY` in the cluster state for this case. So we were "certain" previously but the Danning Kruger effect got us :P 
I think this is a reasonable fix to keep things consistent going forward. The inconsistency with existing indices that had `EMPTY` set on them is annoying but IMO it's also a lesser issue because it has no practical impact for the time being and you could fix it by remounting frozen shards/indices?

Reply:
>  I can't see how we can go from being certain that skipping is correct to being unsure.

We found a bug which we are fixing, that is the reason. We are adjusting the reading side as well for this specific case, but it seems incorrect to keep on using EMPTY for something that should be UNKNOWN.